File: latihanB/python/Memory.py
from Hardware import *

class Memory(Hardware):
    def __init__(self):
        Hardware.__init__(self)
        self.frecuency = ""
        self.memorySize = ""
        self.supportsCuda = "No"

    def setFrecuency(self, frecuency):
        self.frecuency = frecuency

    def getFrecuency(self):
        return self.frecuency
    
    def setMemorySize(self, memorySize):
        self.memorySize = memorySize
    
    def getMemorySize(self):
        return self.memorySize
    
    def setSupportsCuda(self, supportsCuda):
        self.supportsCuda = supportsCuda

    def getSupportsCuda(self):
        return self.supportsCuda

File: latihanB/cpp/Memory.cpp
#include <iostream>
#include <string>
#include "Hardware.cpp"

using namespace std;

class Memory : public Hardware {
    private:
        string frecuency;
        string memorySize;
        string supportsCuda;

    public:
        Memory() {
        }

        Memory(string frecuency, string memorySize, string supportsCuda) {
            this->frecuency = frecuency;
            this->memorySize = memorySize;
            this->supportsCuda = supportsCuda;
        }

        void setFrecuency(string frecuency) {
            this->frecuency = frecuency;
        }

        string getFrecuency() {
            return this->frecuency;
        }

        void setMemorySize(string memorySize) {
            this->memorySize = memorySize;
        }

        string getMemorySize() {
            return this->memorySize;
        }

        void setSupportsCuda(string supportsCuda) {
            this->supportsCuda = supportsCuda;
        }

        string getSupportsCuda() {
            return this->supportsCuda;
        }

        ~Memory() {
        }
};